### PR TITLE
Make GH API URLs configurable

### PR DESCRIPTION
**Pull request by shudgston**
1. Add a `git.api_base` config section to `cirrus.conf` and allow `github_tools` to use it.

```
[github]
api_base = https://api.github.com
```
2. Remove the cirrus package from unit tests to alleviate import confusion and errors when running tests, and to layout the tests directory in a more standard fashion.

3. Modify `tests/unit/github_tools_test.py` to add `GitHubContext` tests.

Others:

* Add `GitHubContextTest` tests
* Refactor `GithubToolsTest` tests
* Remove `data=json.dumps(x)` calls to requests, replace with `json=x`

### CODE REVIEW

File: src/cirrus/configuration.py [hunk 1]
```diff
@@ -337,3 +337,15 @@ def get_chef_auth():
...
+
+def get_github_api_base():
+    """
+    Return the github.api_base URL, or https://api.github.ibm.com if not
+    configured.
+    """
+    try:
+        url = load_configuration()['github']['api_base'].rstrip('/')
+    except KeyError:
+        url = 'https://api.github.ibm.com'
```
Comment (jcounts) [inline src/cirrus/configuration.py:353]: print out some warning message like, "Failed to load api_base from github config, using default: https://api.github.ibm.com"

File: src/cirrus/cirrus_setup.py [hunk 2]
```diff
@@ -10,11 +10,11 @@
 
 from argparse import ArgumentParser
 
-from cirrus.configuration import load_setup_configuration, get_creds_plugin
+from cirrus.configuration import load_setup_configuration, get_github_api_base
 from cirrus.logger import get_logger
 
 LOGGER = get_logger()
-GITHUB_AUTH_URL = "https://api.github.com/authorizations"
+GITHUB_AUTH_URL = get_github_api_base()
```
Comment (jcounts) [inline src/cirrus/cirrus_setup.py]: Looks like this is only used in `create_github_token`. Can we remove the constant and just call `get_github_api_base` there?

File: src/cirrus/cirrus_setup.py [hunk 3]
```diff
@@ -67,13 +67,15 @@ def create_github_token():
     requests for things like pull requests
 
     """
+    authz_url = urljoin(get_github_api_base(), '/authorizations')
```
Comment (jcounts) [inline src/cirrus/cirrus_setup.py:70]: The `/` is needed in `'authorizations'`?

File: src/cirrus/configuration.py [hunk 4]
```diff
@@ -348,4 +350,8 @@ def get_github_api_base():
         url = load_configuration()['github']['api_base'].rstrip('/')
     except KeyError:
         url = 'https://api.github.ibm.com'
+        LOGGER.info(
```
Comment (jcounts) [inline src/cirrus/configuration.py:354]: `.warning` probably makes more sense here

Comment (shudgston) [reply src/cirrus/configuration.py:354]: A warning sounds biased. Its an optional config item.

File: src/cirrus/cirrus_setup.py (hunk 3 above)
Comment (shudgston) [reply src/cirrus/cirrus_setup.py:70]: Yes, see the original value (I screwed this up) https://github.com/cloudant/cirrus/pull/41/commits/f4ff175dd8285623b352adf3d03ba7a22cba848c#diff-5a56a3b8782b6cde3e30a41d7669cc59L17

**Review by jcounts — APPROVED** [review]

**Pull request merged**